### PR TITLE
Get provider_job_id from DelayedJob

### PR DESCRIPTION
**Pull request by kddnewton**
When queueing with DelayedJob, get the id of the job instance and report
it back to ActiveJob as provider_job_id.

### CODE REVIEW

**Comment by jvanbaarsen** [comment]
Should we document somewhere which adapters are returning this ID? Since at the moment only DelayedJob is returning this information. But I can see Sidekiq etc. also setting that var.

**Comment by kddeisz** [comment]
Definitely a good idea. I didn't try the other backends just because I haven't had experience with them, someone else could hopefully add that functionality. Any idea where to document that?

**Comment by kddeisz** [comment]
Fixed the test error, should be good now.

**Comment by kddeisz** [comment]
Just pushed a fix to the branch for the failing test. Don't know why Travis is failing now. Either way, feedback?

File: activejob/test/integration/queuing_test.rb [hunk 1]
```diff
@@ -55,4 +55,11 @@ class QueuingTest < ActiveSupport::TestCase
       skip
     end
   end
+
+  test 'should supply a provider_job_id to DelayedJob' do
+    skip unless adapter_is?(:delayed_job)
+    test_job = TestJob.perform_later @id
+    assert_kind_of Fixnum, test_job.provider_job_id
+  end
+
```
Comment (rafaelfranca) [inline activejob/test/integration/queuing_test.rb]: :scissors: this blank line

**Comment by rafaelfranca** [comment]
Looks good to me.

**Comment by kddeisz** [comment]
Fixed the blank line, should be good now.

**Pull request merged**